tools/code_format/check_format_go_code.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -e
+
+tools="$(dirname "$(dirname "$(realpath "$0")")")"
+root=$(realpath "$tools/..")
+
+cd "$root" || exit 1
+
+gofmt="@go_sdk//:bin/gofmt"
+
+# only get the filenames which not satisfy gofmt
+files=$(find . -name "*.go" -exec bazel run $gofmt -- -l {} \;)
+if [[ $files != "" ]]; then
+  # write changes to original files, so that we can get the changes by git diff.
+  find . -name "*.go" -exec bazel run $gofmt -- -w {} \;
```

Comment:
would it be possible to run this once with a list of files ?

currently its a bit of a log hog in CI as it runs each as a separate bazel job

relatedly its triggering a "WARNING ... @go_sdk//:bin/gofmt is a source file, nothing will be built for it. If you want to build ..." on each invokation

not sure exactly what is causing this - i get the same if i try to build it locally - seems like an upstream issue

either way - if we can just invoke the bin once that will limit the output

Reply:
```console

In tools/code_format/check_format_go_code.sh line 19:
  $GOFMT_BIN -w $go_files
                ^-------^ SC2086 (info): Double quote to prevent globbing and word splitting.

```

i think you will need to use a bash `${ARRAY[@]}` for `go_files`